FlowCal/excel_ui.py

```diff
@@ -379,15 +379,45 @@ def process_beads_table(beads_table,
                 beads_sample_gated = FlowCal.gate.high_low(
                     beads_sample_gated,
                     channels=sc_channels)
-            # Density gating
-            beads_sample_gated, __, gate_contour = FlowCal.gate.density2d(
-                data=beads_sample_gated,
-                channels=sc_channels,
-                gate_fraction=beads_row['Gate Fraction'],
-                xscale='logicle',
-                yscale='logicle',
-                sigma=5.,
-                full_output=True)
+
+            # Scatter gate
+            if 'Scatter Gate' in beads_row.index:
+                # Parse gate method
+                # An empty cell will be seen here as a NaN. Therefore, this has
+                # to be checked for explicitly.
+                if pd.isnull(beads_row['Scatter Gate']):
+                    gate_method = ''
+                else:
+                    gate_method = str(beads_row['Scatter Gate'])
+
+                if gate_method.lower().strip() == 'density':
+                    # Density gate by default
+                    beads_sample_gated, __, gate_contour = FlowCal.gate.density2d(
+                        data=beads_sample_gated,
+                        channels=sc_channels,
+                        gate_fraction=beads_row['Gate Fraction'],
```

Comment:
Seems like you might want to be more robust to the presence / absence of a Gate Fraction column (I'm not actually sure how this code behaves if the Gate Fraction column is omitted).

Also begs the larger question of how should gate parameters be handled in the general case?